homeassistant/components/refoss/switch.py

```diff
@@ -0,0 +1,69 @@
+"""Switch for Refoss."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from refoss_ha.controller.toggle import ToggleXMix
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import COORDINATORS, DISPATCH_DEVICE_DISCOVERED, DISPATCHERS, DOMAIN
+from .entity import RefossEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Refoss device from a config entry."""
+
+    @callback
+    def init_device(coordinator):
+        """Register the device."""
+        device = coordinator.device
+        if not isinstance(device, ToggleXMix):
+            return
+
+        new_entities = []
+        for channel in device.channels:
+            entity = RefossSwitch(coordinator=coordinator, channel=channel)
+            new_entities.append(entity)
+
+        async_add_entities(new_entities)
+
+    for coordinator in hass.data[DOMAIN][COORDINATORS]:
+        init_device(coordinator)
+
+    hass.data[DOMAIN][DISPATCHERS].append(
```

Comment:
Instead of keeping a list of all unsubscribe functions:

```suggestion
    config_entry.async_on_unload(
```